### DISCUSSION TITLE
It dont move

### DISCUSSION THREAD

**Discussion by AndreasSchmidt87** [neuefische/questions#243] [Web] [answered]
Hello everyone, 
My Snake will dont move, but i dont know why can you help me please.
the moveSnake function is for the movement and set Timeout is for repeat the move and regulate the speed.
![Bildschirmfoto 2022-10-11 um 12 24 33](https://user-images.githubusercontent.com/76123735/195066844-36aa655b-e8ee-4c29-819b-f4d3cfb62dee.png)
thanks in advanced.
here the link to the PR https://github.com/AndySch666/capstone-project/pull/23

**Comment by pixelass** [comment, marked as answer]
If you want to repeat a movement you should use `setInterval` instead of `setTimeout`.

BUT: be careful. You need to clean the timeout / interval in the cleanup function of the `useEffect`;

Here's an example: https://codesandbox.io/s/boring-moon-rzdnic?file=/src/App.js

**Reply by pixelass** [reply]
For better performance you could use [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame).

There's a hooks library that offers a hook for this exact use-case: https://github.com/streamich/react-use/blob/master/docs/useRafLoop.md

It also offers a start/stop trigger which is probably very helpful to you.
